### Задание 1

**Из 500 компьютеров  180 принадлежат к 1 партии, 170 – ко второй партии, остальные к третьей. В первой партии 3% брака, во второй — 2%, в третьей – 6%. Случайно выбирается один компьютер. Определить вероятность того, что выбранный компьютер – бракованный.  Округлите полученный ответ до десятитысячных. В качестве разделителя используйте точку.**

In [14]:
#180/500 170/500 150/500    P(Y1)   P(Y2) P(Y3)
#0.03    0.02    0.06       P(X|Y1) P...
#P(X) = 
(180/500)*0.03 + (170/500)*0.02 + (150/500)*0.06   

0.03559999999999999

Ответ: `0.0356`

### Задание 2

**Есть компания с программистами. В ней на языке Java пишут в 3 раза больше человек, чем на языке C++. Программистов на языке C++ в 4 раза больше, чем на Python. Вероятность сделать работу за один день для Java-программиста — 0,85, для программиста на языке C++ 0,9, а для Python программиста 0,8. Найти вероятность того, что программист, выбранный наугад, сделает работу за один день. Ответ округлите до сотых. В качестве десятичного разделителя используйте точку.**

In [15]:
#Java   3*4*x 12/17 0.85
#C++      4*x  4/17 0.9
#Python     x  1/17 0.8
#all     17*x
#             P(yn) P(x|yn)

#P(x) = 
(12/17)*0.85 + (4/17)*0.9 + (1/17)*0.8

0.8588235294117648

Ответ: `0.86`


### Задание 3

**Из 40 снайперов 18 попадает в цель с вероятностью 0,9, 8 — с вероятностью 0,4 и 14 – с вероятностью 0,7. Случайно выбранный снайпер произвел выстрел, поразив цель. Мы хотим вычислить, снайпер из какой группы наиболее вероятно совершил этот выстрел.** 

**В ответ запишите наибольшую вероятность, округленную до тысячных. В качестве разделителя используйте точку.**

In [16]:
#P(x) = 
p = (18/40)*0.9 + (8/40)*0.4 + (14/40)*0.7
#P(y|x) = P(y)*P(x|y)/P(x) =
for py_pxy in [(18/40)*0.9, (8/40)*0.4, (14/40)*0.7]:
    print(py_pxy / p)

0.5547945205479452
0.10958904109589043
0.33561643835616434


первая группа самая вероятная

Ответ: `0.55`

### Задание 4

**Есть 4 кубика. На трех из них окрашена белым половина граней, а на четвертом кубике всего одна грань из шести белая. Наудачу выбранный кубик подбрасывается семь раз. Найти вероятность того, что был выбран четвертый кубик, если при семи подбрасываниях белая грань выпала ровно один раз.  Ответ округлите до тысячных. Используйте точку в качестве десятичного разделителя.**

In [17]:
#0.5 0.5 0.5 1/6                   pi  =  p(x 1 раз | yi)
#0.25                              p(yi)

#bern(n, k, pi) = bern(7, 1, pi) = p(x 1 из 7 раз | yi)

# xx = x 1 из 7 
#  p(y4|xx) =   p(y4) * p(xx| y4) / p(xx)
#  p(xx) = sum( p(yi) * p(xx| yi) )

import scipy.special as scp
def bern(n, k, p):
    return scp.comb(n, k)*(p**k)*(1-p)**(n-k)

pi = [0.5, 0.5, 0.5, 1/6]
py_pxxy = []
for p in pi:
    py_pxxy.append(0.25 * bern(7, 1, p))
py_pxxy[3]/sum(py_pxxy)

0.7042729649328406

Ответ: `0.704`

### Задание 5

**На гитхабе лежит 12 скриптов, написанных аналитиком №1, 20 скриптов — написанных аналитиком №2 и 18 скриптов — написанных аналитиком №3. Вероятность того, что скрипт, написанный аналитиком №1, работает  без ошибок, равна 0,9; для скриптов, написанных аналитиками  №2 и №3, эти вероятности соответственно равны 0,6 и 0,9.**

**Нужно найти вероятность того, что случайно взятый скрипт будет работать без ошибок. В ответ запишите число, округленное до сотых. В качестве десятичного разделителя используйте точку.**

In [18]:
am_y = [12, 20, 18]
py = [x/sum(am_y) for x in am_y]
pxy = [0.9, 0.6, 0.9]

sum(map(lambda x, y: x*y, py, pxy))

0.78

Ответ: `0.78`


### Задание 6

**Известно, что около 60% всех людей в городе болеют заболеванием А. Для определения того, заболел человек или нет, проводится тест, точность которого составляет 99%. Вы получили положительный тест, какова вероятность того, что вы реально больны?**

**Округлите ответ до тысячных, в качестве десятичного раделителя используйте точку.**

In [19]:
# рез теста+ и реальн болезнь
# x            y

py = [0.6, 0.4]
pxy = [0.99, 0.01] #вер пол рез при болезни 
#py1x = ?            вер болезни с пол рез
px = sum([x*y for x, y in zip(py, pxy)])#вер пол рез и для больных и для здоровых
#py1x = 
py[0] * pxy[0] / px 

0.9933110367892977

Ответ: `0.993`


### Задание 7

**Три фабрики производят лампочки. Фабрика А производит 20% всех лампочек, 50% лампочек производится на фабрике В и оставшиеся 30% на фабрике С. 2% лампочек, произведенных на фабрике А, 1% лампочек, произведенных на фабрике В, и 3% лампочек, произведенных на фабрике С, являются бракованными.На рынке наугад выбирается лампочка и оказывается бракованной. Какова вероятность того, что эта лампочка была произведена на заводе B?**

**Ответ округлите до тысячных, в качестве разделителя  используйте точку.**

In [20]:
py = [0.2, 0.5, 0.3]
pxy = [0.02, 0.01, 0.03] 
px = sum([x*y for x, y in zip(py, pxy)])
#py2x = 
py[1] * pxy[1] / px 

0.27777777777777773

Ответ: `0.278`


### Задание 8

**Система РЛС (радиолокационная система) может обнаружить присутствие самолета в 98% случаев. Однако если в зоне действия радара нет самолета, система все равно сообщает (ложно), что самолет присутствует, в 5% случаев. В любой момент времени вероятность того, что самолет находится в зоне действия радара, составляет 7%.Какова вероятность того, что самолет находится в зоне действия радара, учитывая, что на радаре нет самолета?**

**Ответ округлите до десятитысячных, в качестве десятичного разделителя используйте точку.**

In [24]:
import numpy as np
px1y = np.array([0.98, 0.05]) #на радаре есть самолет
px2y = 1 - px1y    #!!на радаре нет самолета
py  = np.array([0.07, 0.93]) #в зоне д-я радара или нет
px2 = sum(py*px2y)
#py1x2 =  
py[0] * px2y[0] / px2

0.0015820996722793552

Ответ: `0.0016`

### Задание 9

**Система РЛС (радиолокационная система) может обнаружить присутствие самолета в 98% случаев. Однако если в зоне действия радара нет самолета, система все равно сообщает (ложно), что самолет присутствует, в 5% случаев. В любой момент времени вероятность того, что самолет находится в зоне действия радара, составляет 7%.Какова вероятность того, что самолет находится в зоне действия радара, учитывая, что радар показывает наличие самолета?**

**Ответ округлите до тысячных, в качестве десятичного разделителя используйте точку.**

In [25]:
px1 = sum(py*px1y)
#py1x1 = 
py[0] * px1y[0] / px1

0.5960034752389227

Ответ: `0.596`

### Задание 10

**Передатчик аварийного локатора самолета (ELT) - это устройство, предназначенное для передачи сигнала в случае аварии. Компания A производит 80% всех ELT на рынке, компания B - 15%, а компания C - остальные 5%. Уровень дефектности ELT, произведенных компанией A, составляет 4%, компанией B - 6%, а компанией C - 9%. Случайно выбранный ELT был протестирован и оказался бракованным. Найдите вероятность того, что он был произведен компанией A.**

**Округлите ответ до тысячных, в качестве десятичного разделителя используйте точку.**

In [26]:
py  = np.array([0.8, 0.15, 0.05])
pxy = np.array([0.04, 0.06, 0.09]) 
px = sum(py*pxy)
#pyx =  
py[0] * pxy[0] / px

0.7032967032967034

Ответ: `0.703`